collectd/files/plugin/contrail_apis.py

```diff
@@ -0,0 +1,138 @@
+INTERVAL = 60
+
+
+def check_state(item, state):
+    return item.getElementsByTagName(
+        "state")[0].childNodes[0].toxml() == state
+
+
+class ContrailApiPlugin(base.Base):
+
+    def __init__(self, *args, **kwargs):
+        super(ContrailApiPlugin, self).__init__(*args, **kwargs)
+        self.plugin = NAME
+        self.session = requests.Session()
+        self.session.mount(
+            'http://',
+            requests.adapters.HTTPAdapter(max_retries=self.max_retries)
+        )
+        self.session.mount(
+            'https://',
+            requests.adapters.HTTPAdapter(max_retries=self.max_retries)
+        )
+        self.urls = {}
+        self.xml_element = {}
+        self.result_type = {}
+        self.state = {}
+
+    def config_callback(self, config):
+        super(ContrailApiPlugin, self).config_callback(config)
+        for node in config.children:
+            self.logger.debug("Got config request for '{}': {} {}".format(
+                node.key.lower(), node.values[0], node.values[1])
+            )
+            if node.key.lower() == "url":
+                self.urls[node.values[0]] = node.values[1]
+            elif node.key.lower() == 'xmlelement':
+                self.xml_element[node.values[0]] = node.values[1]
+            elif node.key.lower() == 'resulttype':
+                self.result_type[node.values[0]] = node.values[1]
+            elif node.key.lower() == 'state':
+                self.state[node.values[0]] = node.values[1]
+
+    def itermetrics(self):
+        for name, url in self.urls.items():
+            self.logger.debug("Requesting {} URL {}".format(
+                name, url)
+            )
+            try:
+                r = self.session.get(url, timeout=self.timeout)
+            except Exception as e:
+                self.logger.error("Got exception for '{}': {}".format(
+                    url, e)
+                )
+                yield {'type_instance': name, 'values': self.FAIL}
+            else:
+                if r.status_code != 200:
+                    self.logger.error(
+                        ("{} ({}) responded with code {} "
+                         "").format(name, url,
+                                    r.status_code))
+                    yield {'type_instance': name, 'values': self.FAIL}
+                else:
+                    try:
+                        self.logger.debug(
+                            "Got response from {}: '{}'"
+                            "".format(url, r.text))
+                        px = xml.dom.minidom.parseString(r.text)
+                        itemlist = px.getElementsByTagName(
+                            self.xml_element[name]
+                        )
+                        if name not in self.result_type:
+                            count = 0
+                            state = None
+                            if name in self.state:
+                                state = self.state[name]
```

Comment:
Or more concisely:

```py
state = self.state.get('name')
```

(`state` will be `None` if `self.state` does not have a key `name`).